solutions/09-11/script.js

```diff
@@ -0,0 +1,97 @@
+function validateCardNumber(value) {
+    return !(/[^0-9-\s]+/.test(value));
+}
+
+function luhnAlghorithm(value) {
+    let nCheck = 0, bEven = false;
+	value = value.replace(/\D/g, "");
+
+	for (let n = value.length - 1; n >= 0; n--) {
+		let cDigit = value.charAt(n),
+			  nDigit = parseInt(cDigit, 10);
+
+		if (bEven && (nDigit *= 2) > 9) nDigit -= 9;
+
+		nCheck += nDigit;
+		bEven = !bEven;
+	}
+	return (nCheck % 10) == 0;
+}
+
+// Card recognition function - first char is reserved for card type
+function whatCard(value) {
+    switch(value[0]) {
+        case '3': {
+            icon.classList.add('fa-cc-amex');
+            break;
+        }
+        case '4': {
+            icon.classList.add('fa-cc-visa');
+            break;
+        }
+        case '5': {
+            icon.classList.add('fa-cc-mastercard');
+            break;
+        }
+        case '6': {
+            icon.classList.add('fa-cc-discover');
+            break;
+        }
+        default: {
+            icon.className = "";
+            icon.classList.add('icon');
+            icon.classList.add('fa');
+        }
+    }
+}
+
+const icon = document.querySelector('#icon');
+const myCard = document.querySelector('#cardNumber');
+const err = document.querySelector("#errCard");
+const owner = document.querySelector('#owner');
+const errOwner = document.querySelector("#errOwner");
+const expDate = document.querySelector('#exp-date');
+const errDate = document.querySelector('#errDate');
+const cvv = document.querySelector('#cvv');
+const errCvv = document.querySelector("#errCvv");
+
+function cardValidation()
+{
+    const val = myCard.value;
+    if (!validateCardNumber(val) || !luhnAlghorithm(val)) {
+        err.textContent="Wrong Card Number";
+    } else {
+        err.textContent = "";
+        whatCard(val);
+    }
+}
+
+function cvvValidation()
+{
+    !(/^[0-9]{3,4}$/.test(cvv.value)) ?
+        errCvv.textContent="Wrong CVV value" :
+        errCvv.textContent = "";
+}
+
+function ownerValidation() {
+    !(/^[a-zA-Z\s]*$/.test(owner.value)) ?
+        errOwner.textContent="Use only letters and spaces" :
```

Comment:
I think it doesn't work

Reply:
![image](https://user-images.githubusercontent.com/36031780/79061979-27081080-7c96-11ea-9439-f6d04f83eb1e.png)
![image](https://user-images.githubusercontent.com/36031780/79061983-32f3d280-7c96-11ea-9290-e404d01c8b2c.png)
I've tested it :P